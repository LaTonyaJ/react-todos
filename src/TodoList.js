import React, {useState} from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList(){
    const [todo, setTodo] = useState([]);

    const addTodo = (newTodo) => {
        setTodo(todo => [...todo, newTodo]);
    }

    const remove = (id) => {
        console.log('removing')
        setTodo(todo => todo.filter(t => t.id !== id));
    }

    return(
        <div>
            <NewTodoForm addTodo={addTodo}/>
            <ul>
                {todo.map(todo => 
                    (<Todo
                     id={todo.id}    
                     key={todo.id} 
                     task={todo.formData} 
                     remove={remove}
                    />))}
            </ul>
        </div>
    );
}

export default TodoList;