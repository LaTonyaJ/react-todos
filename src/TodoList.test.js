import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import TodoList from './TodoList';

//smoke test
it('renders without crashing', function() {
    render(<TodoList />);
});

it('matches snapshot', function() {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("can add a new todo", function() {
    const { getByPlaceholderText, queryByText } = render(<TodoList />);
  
    // no todos yet
    expect(queryByText("wash car")).not.toBeInTheDocument();
  
    const todoInput = getByPlaceholderText("Enter Task");
    const submitBtn = queryByText("Add Todo")
  
    // fill out the form
    fireEvent.change(todoInput, { target: { value: "wash car" }});
    fireEvent.click(submitBtn);
  
    // todo exists!
    expect(queryByText("wash car")).toBeInTheDocument();
  });

  it("can remove todo", function() {
    const { getByPlaceholderText, queryByText } = render(<TodoList />);
  
    // no todos yet
    expect(queryByText("wash car")).not.toBeInTheDocument();
  
    const todoInput = getByPlaceholderText("Enter Task");
    const submitBtn = queryByText("Add Todo")
  
    // fill out the form
    fireEvent.change(todoInput, { target: { value: "wash car" }});
    fireEvent.click(submitBtn);
  
    // todo exists!
    expect(queryByText("wash car")).toBeInTheDocument();

    const remove = queryByText('X');

    fireEvent.click(remove);

    expect(queryByText('wash car')).not.toBeInTheDocument();
  });