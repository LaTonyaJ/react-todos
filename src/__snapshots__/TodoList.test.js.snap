// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h3>
        Todo List:
      </h3>
      <input
        name="task"
        placeholder="Enter Task"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
